### PR TITLE
[15.7] [IDE] Check if a tag is present before adding or removing it

### PR DESCRIPTION
**Pull request by iainx**
Check if a tag is present before adding or removing it from the tag table,
because EndTask can be called multiple times for the same monitor.

Fixes VSTS #591647

### CODE REVIEW

**Comment by GLjackyvaswani** [comment]
Hi Iain,

Is this workitem and associated PR required and action from QA side.

Please revert if we have to work on the same with some more details.

Thanks

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.Components/LogView.cs [hunk 1]
```diff
@@ -553,63 +552,68 @@ void addQueuedUpdate (QueuedUpdate update)
...
 		{
 			var et = new QueuedEndTask (monitor);
 			addQueuedUpdate (et);
 		}
 		
 		protected void UnsafeEndTask (LogViewProgressMonitor monitor)
 		{
-			if (indents.Count > 0 && indents.Pop () != null) {
-				if (monitor != null) {
+			Console.WriteLine ($"Ending task {monitor}");
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.Components/LogView.cs]: Debug code here.

Comment (slluis) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.Components/LogView.cs]: @iainx Can you fix?

**Comment by iainx** [comment]
@slluis fixed. sorry, didn't see the email from GitHub about it

File: main/tests/Ide.Tests/MonoDevelop.Ide.Tests.csproj [hunk 2]
```diff
@@ -54,6 +54,12 @@
       <HintPath>..\..\build\bin\Microsoft.CodeAnalysis.Workspaces.Desktop.dll</HintPath>
       <Private>False</Private>
     </Reference>
+    <Reference Include="Microsoft.CodeAnalysis.Workspaces">
+      <HintPath>..\..\build\bin\Microsoft.CodeAnalysis.Workspaces.dll</HintPath>
+    </Reference>
+    <Reference Include="Microsoft.CodeAnalysis.Workspaces.Desktop">
+      <HintPath>..\..\build\bin\Microsoft.CodeAnalysis.Workspaces.Desktop.dll</HintPath>
+    </Reference>
   </ItemGroup>
```
Comment (mrward) [inline main/tests/Ide.Tests/MonoDevelop.Ide.Tests.csproj]: This looks like the known VS Mac bug has added a whole lot of duplicate references here. We should remove all these extra references in all the .csproj files.

Comment (Therzok) [reply main/tests/Ide.Tests/MonoDevelop.Ide.Tests.csproj]: More info: https://github.com/mono/monodevelop/issues/4381#issuecomment-378595116

**Comment by iainx** [comment]
Argh! Forgot to push a change

**Comment by slluis** [comment]
@iainx please make sure this is also fixed in master.

**Comment by iainx** [comment]
@slluis Cherry-picked to https://github.com/mono/monodevelop/pull/4448

**Pull request merged**